src/relations/postgresql_provider.py

```diff
@@ -207,11 +208,23 @@ def update_endpoints(self, event: DatabaseRequestedEvent = None) -> None:
                 read_only_endpoints,
             )
 
-            # Set connection string URI.
-            self.database_provides.set_uris(
-                relation_id,
-                f"postgresql://{user}:{password}@{self.charm.primary_endpoint}:{DATABASE_PORT}/{database}",
-            )
```

Comment:
Thinking about it more we can most probably just `continue` on line 198 if there's no database or password